crates/services/consensus_module/poa/src/verifier.rs

```diff
@@ -0,0 +1,144 @@
+use anyhow::ensure;
+use fuel_core_chain_config::PoABlockProduction;
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    blockchain::{
+        block::Block,
+        header::BlockHeader,
+        primitives::BlockHeight,
+    },
+    fuel_types::Bytes32,
+    tai64::Tai64N,
+};
+use std::ops::Add;
+
+#[cfg(test)]
+mod tests;
+
+/// The config of the block verifier.
+pub struct Config {
+    /// If the manual block is enabled, skip verification of some fields.
+    pub enabled_manual_blocks: bool,
+    /// This configuration exists because right now consensus module uses the `Tai64::now`
+    /// during the generation of the block. That means if the node(block producer) was
+    /// offline for a while, it would not use the expected block time after
+    /// restarting(it should generate blocks with old timestamps).
```

Comment:
> (it should generate blocks with old timestamps)

Don't follow why we'd want to forge stale timestamps for blocks. Using older timestamps may make the verification logic easier to implement, but the time should always aim to be as current/correct as possible.